sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -764,6 +764,25 @@ class SessionCatalog(
     }
   }
 
+  def lookupTempView(table: String): Option[SubqueryAlias] = {
+    val formattedTable = formatTableName(table)
+    getTempView(formattedTable).map { view =>
+      SubqueryAlias(formattedTable, view)
+    }
+  }
+
+  def lookupGlobalTempView(db: String, table: String): Option[SubqueryAlias] = {
```

Comment:
I think this is safe and I do prefer these methods to a combined `resolveRelation`, but I'm curious why you decided not to use the existing method?

Reply:
The idea here is to clearly separate the resolution of temp view and v1/v2 table, so I'd like to avoid using `resolveRelation` which mixes everything together.

These 2 methods mostly copy-paste code from `resolveRelation`. We can update `resolveRelation` to only resolve v1 tables, but I'd like to do it later as there are many tests calling `resolveRelation` and we need to update them as well.